main.tf

```diff
@@ -7,24 +7,44 @@ data "aws_region" "current" {}
 resource "aws_cognito_user_pool" "main_pool" {
   name = var.pool_name
 
-  username_attributes = var.username_attributes
+  username_attributes        = var.username_attributes
+  sms_authentication_message = var.sms_authentication_message
 
   password_policy {
-    minimum_length    = var.password_minimm_length
-    require_lowercase = var.password_require_lowercase
-    require_uppercase = var.password_require_uppercase
-    require_numbers   = var.password_require_numbers
-    require_symbols   = var.password_require_symbols
+    minimum_length                   = var.password_minimm_length
+    require_lowercase                = var.password_require_lowercase
+    require_uppercase                = var.password_require_uppercase
+    require_numbers                  = var.password_require_numbers
+    require_symbols                  = var.password_require_symbols
+    temporary_password_validity_days = var.temporary_password_validity_days
   }
 
   admin_create_user_config {
     allow_admin_create_user_only = var.allow_admin_create_user_only
+
+    dynamic "invite_message_template" {
+      for_each = var.invite_message_template
```

Comment:
Are multiple `invite_message_template` blocks supported in this resource? If not, maybe `count` is usable instead? Same for the `string_attribute_constraints`.

Reply:
sounds good.

Reply:
couldn't figure out the count but made the object singular and did it a bit different